cpp/include/cudf/detail/utilities/vector_factories.hpp

```diff
@@ -28,6 +30,51 @@
 namespace cudf {
 namespace detail {
 
+/**
+ * @brief Asynchronously construct a `device_uvector` and set all elements to zero.
+ *
+ * @note This function does not synchronize `stream`.
+ *
+ * @tparam T The type of the data to copy
+ * @param size The number of elements in the created vector
+ * @param stream The stream on which to allocate memory and perform the memset
+ * @param mr The memory resource to use for allocating the returned device_uvector
+ * @return A device_uvector containing zeros
+ */
+template <typename T>
+rmm::device_uvector<T> make_zero_device_uvector_async(
```

Comment:
I feel like this naming is a little more intuitive/descriptive. 
```suggestion
rmm::device_uvector<T> make_zeroed_device_uvector_async(
```

Reply:
Done.